CRITICAL_FIXES.py

```
@@ -0,0 +1,357 @@
+AFTER (FIXED):
+"""
+def put_fixed(self, proxy):
+    """Put proxy in pool with correct priority calculation."""
+    is_exceed_time = (proxy.error_rate > self._max_error_rate) or (
+        proxy.avg_resp_time > self._max_resp_time
+    )
+    
+    if proxy.stat['requests'] < self._min_req_proxy:
+        self._newcomers.append(proxy)
+    elif proxy.stat['requests'] >= self._min_req_proxy and is_exceed_time:
+        log.debug('%s:%d removed from proxy pool' % (proxy.host, proxy.port))
```

Ruff (0.11.9)

Undefined name `log`

(F821)